xs/sources/xsMemory.c

```diff
@@ -547,6 +550,21 @@ void* fxGrowChunks(txMachine* the, txSize size)
 	return block;
 }
 
+void fxGrowKeys(txMachine* the, txID theCount) 
+{
+	if (the->keyDelta > 0) {
+		txID keyDelta = (theCount > the->keyDelta) ? theCount : the->keyDelta;
+		txID keyCount = (the->keyCount + keyDelta) - the->keyOffset;
+		txSlot** keyArray = c_realloc(the->keyArray, keyCount * sizeof(txSlot*));
```

Comment:
Note for the future: this uses `c_realloc()` directly, unlike slots/chunks which go through an intermediate `fxAllocateChunks`/`fxAllocateSlots`, which `xsnapPlatform.c` overrides to track in the space meter. So the space used by keys won't be tracked.

This `keyArray` wants to use *re*-alloc, which is not quite as easy to wrap/track. But when we start trying to meter space allocation in earnest, we should consider creating an `fxReallocKeys()`, which can be controlled by the platform, and which can compare the old size (i.e. it can somehow learn the old size, maybe it should accept a third arg with the old size, maybe `the->keyCount`?) and new size, and apply the difference to the meter.

Reply:
Good call! Where should we track this?

Reply:
https://github.com/Agoric/agoric-sdk/issues/7509